src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -66,7 +204,7 @@ public override TypedExpression Transform(TypedExpression ex)
                         QsExpressionKind<TypedExpression, Identifier, ResolvedType>.NewCallLikeExpression(
                             new TypedExpression(
                                 QsExpressionKind<TypedExpression, Identifier, ResolvedType>.NewIdentifier(
```

Comment:
.. and I know that a proper recursion is also part of the next one.

Reply:
Yes in PR #206 .